hastexo/jobs.py

```diff
@@ -90,37 +90,43 @@ def suspend_stack(self, stack):
         Suspend the stack.
 
         """
-        credentials = get_credentials(self.settings, stack.provider)
-        heat_client = self.get_heat_client(credentials)
-
         try:
-            heat_stack = heat_client.stacks.get(stack_id=stack.name)
-        except HTTPNotFound:
-            heat_status = DELETED_STATE
-        else:
-            heat_status = heat_stack.stack_status
+            credentials = get_credentials(self.settings, stack.provider)
+            heat_client = self.get_heat_client(credentials)
 
-        if heat_status in UP_STATES:
-            self.log("Suspending stack [%s]." % stack.name)
+            try:
+                heat_stack = heat_client.stacks.get(stack_id=stack.name)
+            except HTTPNotFound:
+                heat_status = DELETED_STATE
+            else:
+                heat_status = heat_stack.stack_status
 
-            # Suspend stack
-            heat_client.actions.suspend(stack_id=stack.name)
+            if heat_status in UP_STATES:
+                self.log("Suspending stack [%s]." % stack.name)
 
-            # Save status
-            stack.status = SUSPEND_ISSUED_STATE
-            stack.save(update_fields=["status"])
-        else:
-            self.log("Cannot suspend stack [%s] with status [%s]." %
-                     (stack.name, heat_status))
+                # Suspend stack
+                heat_client.actions.suspend(stack_id=stack.name)
 
-            # Schedule for retry, if it makes sense to do so
-            if (heat_status != DELETED_STATE and
-                    'FAILED' not in heat_status):
-                stack.status = SUSPEND_RETRY_STATE
+                # Save status
+                stack.status = SUSPEND_ISSUED_STATE
+                stack.save(update_fields=["status"])
             else:
-                stack.status = heat_status
+                self.log("Cannot suspend stack [%s] with status [%s]." %
+                         (stack.name, heat_status))
+
+                # Schedule for retry, if it makes sense to do so
+                if (heat_status != DELETED_STATE and
+                        'FAILED' not in heat_status):
+                    stack.status = SUSPEND_RETRY_STATE
+                else:
+                    stack.status = heat_status
 
+                stack.save(update_fields=["status"])
+        except Exception as e:
+            # Schedule for retry on any uncaught exception
+            stack.status = SUSPEND_RETRY_STATE
             stack.save(update_fields=["status"])
+            self.log("Error suspending stack [%s]: %s" % (stack.name, str(e)))
```

Comment:
Same as above, perhaps log both stack name and uuid here?

Reply:
Good point, and easy enough to fix.  I'm adding it in.

Reply:
Actually, because this is a catch-all exception, we don't store the UUID in the database, and the most likely exception to trigger this error will be a failure to GET from Heat, we don't have access to the stack UUID here.

Reply:
Gotcha, works for me.